src/main/java/org/bricolages/streaming/filter/TextOp.java

```diff
@@ -65,4 +65,15 @@ String castStringForce(Object value) {
         if (!(value instanceof String)) return null;
         return (String)value;
     }
+
+    final Pattern nullCharsPattern = Pattern.compile("^\\x00+$");
+    String removeHeadNullChars(String str) {
+        if (str == null) return null;
+        // return if first char is not \0
+        if (0 != str.indexOf("\0")) return str;
+        // return null if all chars are \0
+        if (nullCharsPattern.matcher(str).matches()) return null;
+        // remove heading \0(s) if some chars follows
+        return str.replaceFirst("^\\x00+", "");
```

Comment:
1. Why not compile this pattern (while above pattern is compiled)?
2. Replace first, then check the string is empty (and return null if empty).